tests/units/adapters/test_data_saver.py

```diff
@@ -34,6 +39,24 @@ def html_writer(tmpdir):
     return HtmlWriter(output_path)
 
 
+@pytest.fixture(params=['txt', 'html'])
+def template_name_fixture(request) -> str:
+    if request.param == 'txt':
+        return str(TEMPLATE_PATH / 'student_report_template.txt')
+    elif request.param == 'html':
+        return str(TEMPLATE_PATH / 'student_report_template.html')
+
+
+@pytest.mark.parametrize('expected_content',
+                         [
+                             json.load(open(TEMPLATE_PATH_DATA / 'test_data_txt.json'))['expected_content'],
+                             json.load(open(TEMPLATE_PATH_DATA / 'test_data_html.json'))['expected_content']
```

Comment:
Here you should have file name only; reading file content should be performed within test function